kin-devplatform-sdk/src/test/java/kin/devplatform/data/blockchain/BlockchainSourceImplTest.java

```diff
@@ -144,47 +148,41 @@ public void extract_order_id() {
 	}
 
 	@Test
-	public void send_transaction_succeeded() throws InterruptedException {
+	public void send_transaction_succeeded() throws InterruptedException, OperationFailedException {
 		String toAddress = "some_pub_address";
 		final BigDecimal amount = new BigDecimal(10);
 		final String orderID = "someID";
-		final String transactionID = "transactionID";
-
-		Request<TransactionId> transactionRequest = mock(Request.class);
-		ArgumentCaptor<ResultCallback<TransactionId>> resultCallbackArgumentCaptor =
-			forClass(ResultCallback.class);
-		when(kinAccount.sendTransaction(any(String.class), any(BigDecimal.class), any(String.class)))
-			.thenReturn(transactionRequest);
 
+		when(kinAccount.sendTransactionSync(anyString(), (BigDecimal) any(), anyString())).thenReturn(
+			new TransactionId() {
+				@Override
+				public String id() {
+					return "transactionID";
+				}
+			});
 		blockchainSource.setAppID(APP_ID);
-		blockchainSource.sendTransaction(toAddress, amount, orderID, "offerID", OfferType.SPEND);
-		verify(transactionRequest).run(resultCallbackArgumentCaptor.capture());
-		resultCallbackArgumentCaptor.getValue().onResult(new TransactionId() {
-			@Override
-			public String id() {
-				return transactionID;
-			}
-		});
+		OpenOrder openOrder = new OpenOrder();
+		openOrder.setId(orderID);
+		openOrder.setOfferId("offerID");
+		openOrder.setOfferType(OfferType.SPEND);
+		blockchainSource.sendTransaction(toAddress, amount, openOrder);
 		verify(eventLogger).send(any(SpendTransactionBroadcastToBlockchainSucceeded.class));
 	}
 
 	@Test
-	public void send_transaction_failed() {
+	public void send_transaction_failed() throws OperationFailedException {
 		String toAddress = "some_pub_address";
 		BigDecimal amount = new BigDecimal(10);
 		final String orderID = "someID";
 
-		Request<TransactionId> transactionRequest = mock(Request.class);
-		ArgumentCaptor<ResultCallback<TransactionId>> resultCallbackArgumentCaptor =
-			forClass(ResultCallback.class);
-		when(kinAccount.sendTransaction(any(String.class), any(BigDecimal.class), any(String.class)))
-			.thenReturn(transactionRequest);
+		final OperationFailedException exception = new OperationFailedException("failed");
+		when(kinAccount.sendTransactionSync(any(String.class), any(BigDecimal.class), any(String.class)))
+			.thenThrow(exception);
 
 		blockchainSource.setAppID(APP_ID);
-		blockchainSource.sendTransaction(toAddress, amount, orderID, "offerID", OfferType.SPEND);
-		verify(transactionRequest).run(resultCallbackArgumentCaptor.capture());
-
-		final Exception exception = new Exception("failed");
+		OpenOrder openOrder = new OpenOrder();
```

Comment:
OpenOrder can function without id and type? Why not pass these parameters in the constructor?